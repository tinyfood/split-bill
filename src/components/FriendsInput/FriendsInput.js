import React from 'react'
import Slider, { Range } from 'rc-slider'
import { connect } from 'react-redux'
import { editFriends } from '../../actions/split'

const FriendsInput = ({ dispatch }) => {
  return (
    <div>
      <Slider
        min={2}
        max={8}
        step={1}
        handleStyle={{
          borderColor: 'blue',
          height: 28,
          width: 28,
          marginLeft: -14,
          marginTop: -9,
          backgroundColor: 'black'
        }}
        railStyle={{ backgroundColor: 'red', height: 10 }}
        onChange={value => {
          dispatch(editFriends(value))
        }}
      />
    </div>
  )
}

export default connect()(FriendsInput)
