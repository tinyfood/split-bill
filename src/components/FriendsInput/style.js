import styled from 'styled-components'

export const FriendsInputStyled = styled.div`
  position: relative;
  margin: 38px 0;
`

export const FriendsCount = styled.div`
  position: absolute;
  top: 4px;
  right: 10px;
  font-size: 18px;
  font-weight: 700;
  color: ${props => (props.white ? '#ffffff' : '#4e4e4e')};
`

export const TrackStyle = {
  position: 'relative',
  top: '-30px',
  backgroundColor: '#2ecc71',
  height: '30px',
  borderTopLeftRadius: '4px',
  borderBottomLeftRadius: '4px',
  transition: 'all .075s linear',
  cursor: 'pointer'
}

export const RailStyle = {
  height: '30px',
  backgroundColor: 'rgba(239, 237, 232, 0.5)',
  borderRadius: '4px',
  transition: 'all .075s linear',
  cursor: 'pointer'
}

export const handleStyle = {
  position: 'relative',
  top: '-70px',
  height: 54,
  width: 8,
  backgroundColor: '#2ecc4a',
  borderRadius: '2px',
  outline: 'none',
  transition: 'all .075s linear',
  cursor: 'pointer'
}
