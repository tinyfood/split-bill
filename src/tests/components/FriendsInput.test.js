import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { FriendsInput } from '../../components/FriendsInput/FriendsInput'

test('should render FriendsInput correctly', () => {
  const wrapper = shallow(<FriendsInput />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
