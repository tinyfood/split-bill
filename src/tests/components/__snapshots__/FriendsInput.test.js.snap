// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FriendsInput correctly 1`] = `
<styled.div>
  <ComponentEnhancer(undefined)
    activeDotStyle={Object {}}
    className=""
    disabled={false}
    dotStyle={Object {}}
    dots={false}
    handle={[Function]}
    handleStyle={
      Object {
        "backgroundColor": "#2ecc4a",
        "borderRadius": "2px",
        "cursor": "pointer",
        "height": 54,
        "outline": "none",
        "position": "relative",
        "top": "-70px",
        "transition": "all .075s linear",
        "width": 8,
      }
    }
    included={true}
    marks={Object {}}
    max={6}
    min={1}
    onAfterChange={[Function]}
    onBeforeChange={[Function]}
    onChange={[Function]}
    prefixCls="rc-slider"
    railStyle={
      Object {
        "backgroundColor": "rgba(239, 237, 232, 0.5)",
        "borderRadius": "4px",
        "cursor": "pointer",
        "height": "30px",
        "transition": "all .075s linear",
      }
    }
    step={1}
    style={
      Object {
        "height": "50px",
      }
    }
    trackStyle={
      Object {
        "backgroundColor": "#2ecc71",
        "borderBottomLeftRadius": "4px",
        "borderTopLeftRadius": "4px",
        "cursor": "pointer",
        "height": "30px",
        "position": "relative",
        "top": "-30px",
        "transition": "all .075s linear",
      }
    }
    vertical={false}
  />
  <PersonsIcon />
  <styled.div
    white={false}
  />
</styled.div>
`;
